server-spi/src/main/java/org/keycloak/models/OTPPolicy.java

```diff
@@ -59,15 +60,24 @@ public OTPPolicy() {
     }
 
     public OTPPolicy(String type, String algorithm, int initialCounter, int digits, int lookAheadWindow, int period) {
+        this(type, algorithm, initialCounter, digits, lookAheadWindow, period, DEFAULT_IS_REUSABLE);
+    }
+
+    public OTPPolicy(String type, String algorithm, int initialCounter, int digits, int lookAheadWindow, int period, boolean isCodeReusable) {
         this.type = type;
         this.algorithm = algorithm;
         this.initialCounter = initialCounter;
         this.digits = digits;
         this.lookAheadWindow = lookAheadWindow;
         this.period = period;
+        this.isCodeReusable = isCodeReusable;
     }
 
     public static OTPPolicy DEFAULT_POLICY = new OTPPolicy(OTPCredentialModel.TOTP, HmacOTP.HMAC_SHA1, 0, 6, 1, 30);
+    public static final boolean DEFAULT_IS_REUSABLE = true;
```

Comment:
Will be the new behavior the default one, or must the admin explicitly define that those OTPs cannot be reused?

Reply:
Agree :)